platforms/access-bridge/src/server.ts

```diff
@@ -1,123 +1,106 @@
-import { Server as HTTPServer, IncomingMessage, ServerResponse, createServer } from 'http';
+import { Server as HTTPServer } from 'http';
 
-import { EndpointHandler } from './endpoints.js';
-import { InternalError, MethodNotAllowedError, NotFoundError, AccessBridgeError, sendErrors } from './errors.js';
-import { ALLOWED_REQUEST_METHODS, RequestMethod } from './http.js';
+import express, { Express, Request, Response, NextFunction } from 'express';
+import cors from 'cors';
 
+import { AccessBridgeError, ErrorDefinitions, sendErrors } from './errors.js';
+
+/**
+ * Server class that initializes and manages an Express application with error handling.
+ */
 export class Server {
+  private app: Express;
   private httpServer: HTTPServer | null;
   private address: string;
   private port: number;
-  private endpointsHandler: EndpointHandler;
 
-  constructor(address: string, port: number, endpointsHandler: EndpointHandler) {
+  /**
+   * Creates an instance of the Server class.
+   * @param address - Address to bind the server to
+   * @param port - Port to bind the server to
+   * @param registerEndpoints - Function to register routes and endpoints
+   */
+  constructor(address: string, port: number, registerEndpoints: (app: Express) => void) {
+    this.app = express();
     this.httpServer = null;
     this.address = address;
     this.port = port;
-    this.endpointsHandler = endpointsHandler;
+    this.initialize(registerEndpoints);
+  }
+
+  /**
+   * Initializes the server with middleware and endpoints.
+   * @param registerEndpoints - Function to register routes and endpoints
+   */
+  private initialize(registerEndpoints: (app: Express) => void) {
+    // Middleware to enable Cross-Origin Resource Sharing (CORS)
+    this.app.use(cors());
+
+    // Middleware for parsing JSON request bodies
+    this.app.use(express.json());
+
+    // Register custom endpoints
+    registerEndpoints(this.app);
+
+    // Handle 404 Not Found errors
+    this.app.use(this.notFoundErrorHandler);
+
+    // Global error handling middleware
+    this.app.use(this.globalErrorHandler);
   }
 
+  /**
+   * Middleware to handle 404 Not Found errors.
+   */
+  private notFoundErrorHandler = (req: Request, res: Response, next: NextFunction) => {
+    sendErrors(res, ErrorDefinitions.NotFoundError.create());
+    return;
+  };
+
+  /**
+   * Global error handler middleware for the server.
+   */
+  private globalErrorHandler = (err: unknown, req: Request, res: Response, next: NextFunction) => {
+    if (err instanceof AccessBridgeError) {
```

Comment:
Can you put the 'isJwError' logic here instead of having it in each try-catch?

Reply:
But then the error instance will not be AccessBridgeError and we will not be able to sendErrors here as they will not be handled. Using isJwError is possible but we still need to handle them before sending. 
Also why we wanna lose the service and controller error level handling. This suggestion also means just passing the errors from service level to controller and from controller to here.

Reply:
Generally the idea with most of these request / response pipeline frameworks is to use the pipeline to your advantage so there can be a single place that 'universal' things happen, instead of having to duplicate it for each endpoint. This feels to me like one of those types of places, but if it's a difficult refactor we can leave it for a later date when we have time to look at it more.

Reply:
This is now applied 😅